packages/lib/gpc/src/gpcChecks.ts

```diff
@@ -157,6 +157,12 @@ function checkProofObjConfig(
     checkProofEntryConfig(`${nameForErrorMessages}.${entryName}`, entryConfig);
     nEntries++;
   }
+  if (objConfig.signerPublicKey !== undefined) {
+    checkProofEntryConfig(
```

Comment:
I think this needs some different behavior to know whether it's checking a real or virtual entry.  E.g. it shouldn't allow isOwnerID in a virtual entry.  I guess it's optional, given that we don't check elsewhere for extra/unknown fields in config, though canonicalization will end up removing them.I think that's worth explicitly checking and ruling out this specific case since it'll be an easy mistake for users to make.

Checking for extra fields is probably something we should start doing at some point, since in general extra fields could indicate the user is using an incompatible version of config.  It's likely to be tedious, though, so I'm hoping to investigate a framework like Zod to do the checking for us before making that general change.